lints/lint_subject_dn_not_printable_characters.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * ZLint Copyright 2017 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"encoding/asn1"
+	"log"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type subjectDNNotPrintableCharacters struct{}
+
+func (l *subjectDNNotPrintableCharacters) Initialize() error {
+	return nil
+}
+
+func (l *subjectDNNotPrintableCharacters) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func (l *subjectDNNotPrintableCharacters) Execute(c *x509.Certificate) *LintResult {
+	rdnSequence := util.RawRDNSequence{}
+	rest, err := asn1.Unmarshal(c.RawSubject, &rdnSequence)
+	if err != nil {
+		return &LintResult{Status: Fatal}
+	}
+	if len(rest) > 0 {
+		return &LintResult{Status: Fatal}
+	}
+
+	for _, attrTypeAndValueSet := range rdnSequence {
+		for _, attrTypeAndValue := range attrTypeAndValueSet {
+			for _, byte := range attrTypeAndValue.Value.Bytes {
+				if byte < 32 {
+					log.Println(attrTypeAndValue.Value.Tag, "bad character:", byte)
+					return &LintResult{Status: Error}
+				}
+				if byte >= 0x7F && byte <= 0x9F {
```

Comment:
Can we use either decimal or hex but not both? Just for readability.

Reply:
Sure. I have made that change.